test/box-luatest/shutdown_test.lua

```diff
@@ -74,6 +75,35 @@ g_crash.test_shutdown_during_snapshot_on_signal = function(cg)
     t.assert_equals(status.exit_code, 0)
 end
 
+-- Test shutdown when new iproto connection is accepted but
+-- not yet fully established.
+g_crash.test_shutdown_during_new_connection = function(cg)
+    local script = [[
+        local net = require('net.box')
+        local fiber = require('fiber')
+
+        local sock = 'unix/:./iproto.sock'
+        box.cfg{listen = sock}
+        local cond = fiber.cond()
+        local in_trigger = false
+        box.session.on_connect(function()
+            in_trigger = true
+            cond:signal()
+            fiber.sleep(1)
+        end)
+        net.connect(sock, {wait_connected = false})
+        while not in_trigger do
+            cond:wait()
+        end
+        os.exit()
+    ]]
+    -- Inherit environment to inherit ASAN suppressions.
+    local env = table.copy(os.environ())
```

Comment:
Nit: `table.copy is useless`.

Reply:
Fixed. Removed passing env to `justrun` altogether as it inherit it anyway. 

```diff
diff --git a/test/box-luatest/shutdown_test.lua b/test/box-luatest/shutdown_test.lua
index cac0bfe53..fd02c1a28 100644
--- a/test/box-luatest/shutdown_test.lua
+++ b/test/box-luatest/shutdown_test.lua
@@ -97,9 +97,7 @@ g_crash.test_shutdown_during_new_connection = function(cg)
         end
         os.exit()
     ]]
-    -- Inherit environment to inherit ASAN suppressions.
-    local env = table.copy(os.environ())
-    local result = justrun.tarantool(cg.workdir, env, {'-e', script},
+    local result = justrun.tarantool(cg.workdir, {}, {'-e', script},
                                      {nojson = true, quote_args = true})
     t.assert_equals(result.exit_code, 0)
 end
```